db/tables.py

```diff
@@ -14,15 +16,17 @@ def create_string_column_table(name, schema, column_names, engine):
     return table
 
 
-def create_mathesar_table(name, schema, columns, engine):
+def create_mathesar_table(name, schema, columns, engine, metadata=None):
     """
     This method creates a Postgres table in the specified schema using the
     given name and column list.  It adds internal mathesar columns to the
     table.
     """
-    columns = init_mathesar_table_column_list_with_defaults(columns)
+    columns = col.init_mathesar_table_column_list_with_defaults(columns)
     schemas.create_schema(schema, engine)
-    metadata = MetaData(bind=engine)
+    if metadata is None:
+        metadata = MetaData(bind=engine, schema=schema)
+    metadata.reflect()
```

Comment:
Why add this here? Please add a comment.

Reply:
It lets you create tables in a metadata containing other tables; necessary for joining, splitting, and so on.  Basically, we need a shared context for operations using more than one table.

The reflection step specifically takes the "this table already exists" error out of the DB layer, and puts it where SQLAlchemy will notice it without sending an error-generating request to the DB.

I'll comment in the code, too.